internal/reader/rmn_home.go

```diff
@@ -0,0 +1,329 @@
+	"fmt"
+	"math/big"
+	"sync"
+	"time"
+
+	mapset "github.com/deckarep/golang-set/v2"
+
+	"github.com/smartcontractkit/chainlink-ccip/pkg/consts"
+
+	"github.com/smartcontractkit/chainlink-ccip/internal/reader/contractreader"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/logger"
+	"github.com/smartcontractkit/chainlink-common/pkg/services"
+	"github.com/smartcontractkit/chainlink-common/pkg/types"
+	cciptypes "github.com/smartcontractkit/chainlink-common/pkg/types/ccipocr3"
+	"github.com/smartcontractkit/chainlink-common/pkg/types/query/primitives"
+
+	rmntypes "github.com/smartcontractkit/chainlink-ccip/commit/merkleroot/rmn/types"
+)
+
+const (
+	expectedRMNHomeConfigs = 2   // RMNHome contract should have 2 configs, primary and secondary
+	rmnMaxSizeCommittee    = 256 // bitmap is 256 bits making the max committee size 256
+)
+
+type RMNHome interface {
+	// GetRMNNodesInfo gets the RMNHomeNodeInfo for the given configDigest
+	GetRMNNodesInfo(configDigest cciptypes.Bytes32) ([]rmntypes.RMNHomeNodeInfo, error)
+	// IsRMNHomeConfigDigestSet checks if the configDigest is set in the RMNHome contract
+	IsRMNHomeConfigDigestSet(configDigest cciptypes.Bytes32) bool
+	// GetMinObservers gets the minimum number of observers required for each chain in the given configDigest
+	GetMinObservers(configDigest cciptypes.Bytes32) (map[cciptypes.ChainSelector]uint64, error)
+	// GetOffChainConfig gets the offchain config for the given configDigest
+	GetOffChainConfig(configDigest cciptypes.Bytes32) (cciptypes.Bytes, error)
+	services.Service
+}
+
+type rmnHomeState struct {
+	primaryConfigDigest   cciptypes.Bytes32
+	secondaryConfigDigest cciptypes.Bytes32
+	rmnHomeConfig         map[cciptypes.Bytes32]rmntypes.RMNHomeConfig
+}
+
+// RmnHomePoller polls the RMNHome contract for the latest RMNHomeConfigs
+// It is running in the background with a polling interval of pollingDuration
+type RmnHomePoller struct {
+	wg                   sync.WaitGroup
+	stopCh               services.StopChan
+	sync                 services.StateMachine
+	mutex                *sync.RWMutex
+	contractReader       contractreader.ContractReaderFacade
+	rmnHomeBoundContract types.BoundContract
+	lggr                 logger.Logger
+	rmnHomeState         rmnHomeState
+	failedPolls          uint
+	pollingDuration      time.Duration // How frequently the poller fetches the chain configs
+}
+
+func NewRMNHomePoller(
+	contractReader contractreader.ContractReaderFacade,
+	rmnHomeBoundContract types.BoundContract,
+	lggr logger.Logger,
+	pollingInterval time.Duration,
+) RMNHome {
+	return &RmnHomePoller{
+		stopCh:               make(chan struct{}),
+		contractReader:       contractReader,
+		rmnHomeBoundContract: rmnHomeBoundContract,
+		rmnHomeState:         rmnHomeState{},
+		mutex:                &sync.RWMutex{},
+		failedPolls:          0,
+		lggr:                 lggr,
+		pollingDuration:      pollingInterval,
+	}
+}
+
+func (r *RmnHomePoller) Start(ctx context.Context) error {
```

Comment:
Where do we expect to call this?

Reply:
This should be similar to HomeChain. Expose RMNHome and New function in [pkg](https://github.com/smartcontractkit/chainlink-ccip/blob/631f15febe7ccea6b0648653b7fe1b5c6d4f0167/pkg/reader/home_chain.go#L22-L22). Then use them from [delegate](https://github.com/smartcontractkit/chainlink/blob/e6c11f2353e0bda76c4c60d700c808942f88b5de/core/capabilities/ccip/delegate.go#L158-L158).

Reply:
No, this won't be like HomeChain. We will have a separate RMNHomeReader for each plugin instance.

Reply:
This would be expcted to be called in the commit `NewPluginFactory`. We would also need to call `Start` after it.

It's going to be in my next PR, I just need to feature flag it: https://github.com/smartcontractkit/chainlink-ccip/pull/140

Reply:
And `Stop` will be called by the plugin, in the `Close` method?

Reply:
Correct